exporters/trace/ocagent/src/test/java/io/opencensus/exporter/trace/ocagent/FakeClientCalls.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.trace.ocagent;
+
+import io.grpc.ClientCall;
+import io.grpc.Metadata;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/** Fake implementations of {@link io.grpc.stub.ClientCalls}. Used in unit tests. */
+final class FakeClientCalls {
+
+  private FakeClientCalls() {}
+
+  // No-op implementation of ClientCall. ClientCall cannot be mocked with annotation:
+  // @DoNotMock("Use InProcessServerBuilder and make a test server instead").
```

Comment:
`InProcessServerBuilder` is more for integration tests. I think it's an overkill to start a real server and create real channels to just test the handler functionalities. See https://github.com/grpc/grpc-java/blob/b7c3d276c9a723d37cd263912461c77485697fef/core/src/main/java/io/grpc/inprocess/InProcessServerBuilder.java#L50-L57.

Reply:
I would like to have that, it is better to test with more real code. I know we can argue about unit-test vs integration test but I think following what gRPC proposes is fine.

Reply:
@zhangkun83 what do you suggest here?

Reply:
In-process is recommended for unit tests. Mocking or faking Channel/ClientCall should be avoided, because the semantics are non-trivial and it's easy to implement it incorrectly thus end up verifying wrong behavior. For example, in reality you may forget to call `ClientCall.request()` and your fake ClientCall may not require it either, thus your test would still pass, but it will not work with a real ClientCall. Such error would be caught if tested with an in-process channel